master/internal/resourcemanagers/resource_pool.go

```diff
@@ -381,36 +381,136 @@ func (rp *ResourcePool) receiveAgentMsg(ctx *actor.Context) error {
 	return nil
 }
 
-func (rp *ResourcePool) moveJob(msg job.MoveJob) error {
-	// REMOVEME
-	fmt.Println(rp.config.PoolName, "moveJob: ", msg)
-	rp.queuePositions[job.TailAnchor] = initalizeQueuePosition(time.Now())
-	newPos, rebalance, err := computeNewJobPos(msg, rp.queuePositions)
-	if err != nil {
-		return err
+func (rp *ResourcePool) moveJob(
+	ctx *actor.Context, jobID model.JobID, anchorID model.JobID, aheadOf bool,
+) error {
+	if rp.config.Scheduler.GetType() != priorityScheduling {
+		return nil
+	}
+
+	if anchorID == "" || jobID == "" || anchorID == jobID {
+		return nil
+	}
+
+	if _, ok := rp.queuePositions[jobID]; !ok {
+		return nil
+	}
+
+	groupAddr, ok := rp.IDToGroupActor[jobID]
+	if !ok {
+		return job.ErrJobNotFound(jobID)
 	}
-	rp.queuePositions[msg.ID] = newPos
-	// condiontally check if rebalancing is needed.
-	if rebalance {
-		adjustPriorities(&rp.queuePositions)
+
+	if _, ok := rp.queuePositions[anchorID]; !ok {
+		return job.ErrJobNotFound(anchorID)
+	}
+
+	prioChange, secondAnchor, anchorPriority := rp.findAnchor(jobID, anchorID, aheadOf)
+
+	check.Panic(check.True(secondAnchor != ""))
+
+	if secondAnchor == jobID {
+		return nil
+	}
+
+	if prioChange {
+		resp := ctx.Self().System().AskAt(job.JobsActorAddr, job.SetJobPriority{ID: jobID, Priority: anchorPriority})
+		if resp.Error() != nil {
+			return resp.Error()
+		}
+		if needMove(
+			rp.queuePositions[jobID],
+			rp.queuePositions[anchorID],
+			rp.queuePositions[secondAnchor],
+			aheadOf,
+		) {
+			return nil
+		}
 	}
 
-	if err := rp.persist(); err != nil {
+	msg, err := rp.queuePositions.SetJobPosition(jobID, anchorID, secondAnchor)
+
+	if err != nil {
 		return err
 	}
-	// actors.NotifyAfter(ctx, 0, schedulerTick{}) ?
+
+	ctx.Tell(groupAddr, msg)
+
 	return nil
 }
 
+func (rp *ResourcePool) findAnchor(jobID model.JobID, anchorID model.JobID, aheadOf bool) (bool, model.JobID, int) {
+	var secondAnchor model.JobID
+	targetPriority := 0
+	anchorPriority := 0
+	anchorIdx := 0
+	anchorPos, _ := rp.queuePositions[anchorID]
+	prioChange := false
+
+	sortedReqs := sortTasksWithPosition(rp.taskList, rp.groups, rp.queuePositions, false)
+
+	for i, req := range sortedReqs {
+		if req.JobID == jobID {
+			targetPriority = *rp.groups[req.Group].priority
+		} else if req.JobID == anchorID {
+			anchorPriority = *rp.groups[req.Group].priority
+			anchorIdx = i
+		}
+	}
+
+	if aheadOf {
+		if anchorIdx == 0 {
+			secondAnchor = job.HeadAnchor
+		} else {
+			secondAnchor = sortedReqs[anchorIdx-1].JobID
+			if rp.queuePositions[secondAnchor].GreaterThanOrEqual(anchorPos) {
+				secondAnchor = job.HeadAnchor
+			}
```

Comment:
maybe i'm just dense, i dont get how this case could ever happen if we're itering sortedReqs and successfully pulling the req prior to the anchor. same for the other case below. what am i missing?

Reply:
This is supposed to cover the case where the priority of the anchors might be different in addition to the queue positions. For example, if the first anchor has priority 42 & position 10, but the second anchor has priority 41 & position 20. Then the second anchor would be sorted before the first anchor, even though it's position is higher (smaller is better).

Reply:
oh. why not just if `rp.groups[secondAnchorReq.Group].priority < &anchorPriority` then. what if the first anchor is priority 42 and position 20 but the second is priority 41 and position 10. seems we should still use the head anchor not the second anchor we've found?

Reply:
Hmm I think using decimal comparisons is the least ugly way to do it. Since the list was already sorted, we know that the second anchor is right next to the first anchor. Since we promote or demote the priority of the target job to be the same as the anchor, we just need a queue position relative to the first anchor too. Ill think about this a bit more

Reply:
>  what if the first anchor is priority 42 and position 20 but the second is priority 41 and position 10. seems we should still use the head anchor not the second anchor we've found?

sorry for being dense. what about this case then? is there some invariant upheld elsewhere that prevents this?